tests/native_api_tests/test_api_model.cpp

```diff
@@ -39,6 +39,17 @@ void check_exception(PowerGridError const& e, PGM_ErrorCode const& reference_err
     std::string const err_msg{e.what()};
     CHECK(err_msg.find(reference_err_msg) != std::string::npos);
 }
+
+template <typename Func, class... Args>
+void error_handler(Func func, PGM_ErrorCode const& reference_error, std::string_view reference_err_msg,
+                   Args&&... args) {
+    try {
+        func(std::forward<Args>(args)...);
+        FAIL("Exception was not thrown when it was supposed to.");
```

Comment:
Sorry for the very late new addition but i felt like it had to exist but i couldn't find it earlier. Now i actually did: just like there's `doctest::Approx` for floating points, there's also `doctest::Contains` for strings (https://github.com/doctest/doctest/blob/master/doc/markdown/assertions.md#string-containment). Maybe this solves the issue with `CHECK_THROWS_WITH_AS`, e g : `CHECK_THROWS_WITH_AS(model.update(update_dataset), doctest::Contains("ID not found"), PowerGridRegularError)`

Not sure if this is all correct but you get the picture. If it works: great, if it doesn't: also ok